### PR TITLE
update cadvisor to fix num CPU detection

### LINKED ISSUE #89859: Zero cpu's (cpu: 0) reported for nodes on K8s built on "master"
**What happened**:

When using K8s built on the master branch `v1.19.0-alpha.1.295+873fe1478a7984` the number of cpu's is 0 (zero);

```
# kubectl get node vm-002 -o json | jq .status.capacity
{
  "cpu": "0",
  "ephemeral-storage": "8191416Ki",
  "hugepages-2Mi": "0",
  "memory": "1531128Ki",
  "pods": "110"
}
```
Output from `kubectl describe node vm-002` [describe-node.txt](https://github.com/kubernetes/kubernetes/files/4432234/describe-node.txt)

**What you expected to happen**:

On `v1.18.0` and all others down to `v1.9.x` "cpu: 2" is reported, and the same is expected.

**How to reproduce it (as minimally and precisely as possible)**:

* Install from scratch with a K8s build from master
* Make `kubectl describe node <name>`

**Anything else we need to know?**:

I install "from scratch" on a minimal system. I have missed things in the past with `cgroups` if that can be a problem this time also(?).

I build the kernel myself and have tested with (with equal result);

* linux-4.20.7
* linux-5.6

I run K8s on kvm VM's with `-smp 2` which gives 2 (virtual) cores 
[proc-cpuinfo.txt](https://github.com/kubernetes/kubernetes/files/4432248/proc-cpuinfo.txt)

I discovered the problem when CNI-plugins; calico, flannel and weave stopped working with
```
 Warning  FailedScheduling  <unknown>  default-scheduler  0/4 nodes are available: 4 Insufficient cpu.
```

**Environment**:

- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.0-alpha.1.295+873fe1478a7984", GitCommit:"873fe1478a79841349aef6ed4b6907dcb7a179b1", GitTreeState:"clean", BuildDate:"2020-04-03T11:21:42Z", GoVersion:"go1.14", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.0-alpha.1.295+873fe1478a7984", GitCommit:"873fe1478a79841349aef6ed4b6907dcb7a179b1", GitTreeState:"clean", BuildDate:"2020-04-03T11:21:26Z", GoVersion:"go1.14", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: Kvm VMs, no cloud provider
- OS (e.g: `cat /etc/os-release`): Own built BusyBox based system
- Kernel (e.g. `uname -a`):
```
Linux vm-002 4.20.7 #2 SMP Sat Apr 4 11:53:05 CEST 2020 x86_64 GNU/Linux
```
- Install tools: "from scratch"
- Network plugin and version (if this is a network-related bug): Severak;
  Working; cilium. Not working; calico, flannel, weave
- Others:

/sig node

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: fixes CPU detection on platforms with SMT disabled. currently kubelet / cadvisor will detect 0 cores (!)

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/89859

**Special notes for your reviewer**:

fixes are from: https://github.com/google/cadvisor/pull/2579 & https://github.com/google/cadvisor/pull/2567

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/cc @dims @liggitt

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -72,14 +74,14 @@ require (
...
-	github.com/mrunalp/fileutils v0.0.0-20171103030105-7d4729fb3618
+	github.com/mrunalp/fileutils v0.0.0-20200520151820-abd8a0e76976
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822
 	github.com/mvdan/xurls v1.1.0
 	github.com/onsi/ginkgo v1.11.0
 	github.com/onsi/gomega v1.7.0
 	github.com/opencontainers/go-digest v1.0.0-rc1
-	github.com/opencontainers/runc v1.0.0-rc10
-	github.com/opencontainers/selinux v1.3.3
+	github.com/opencontainers/runc v1.0.0-rc90.0.20200602184820-0853956d235d
```
Comment (liggitt) [inline go.mod:83]: jumping between tags is o_O, but I guess we don't actually vendor any code from runc...

**Comment by liggitt** [comment]
make sure `hack/lint-dependencies.sh` is happy... several transitive deps appear to be pinned to versions older than the selected minimum version

File: go.mod (hunk 1 above)
Comment (liggitt) [reply go.mod:83]: I take it back... we're pinning runc older than its preferred version, which we can't do

**Comment by BenTheElder** [comment]
working on making lint happy

**Comment by dims** [comment]
@BenTheElder already in progress in https://github.com/kubernetes/kubernetes/pull/91366

waiting on @jingyih https://github.com/kubernetes/kubernetes/issues/91266

We may even need the fix for to land https://github.com/etcd-io/etcd/issues/11992

**Comment by BenTheElder** [comment]
... this is getting a bit ridiculous.

we should not leave ourselves in a state where systems with SMT disabled are broken for > 1 month, this really should have been a revert in cAdvisor / revert of the cAdvisor bump instead of "rolling forward" for > 1 month.

**Comment by BenTheElder** [comment]
discussed in code organization meeting, this is not going to work as-is.

**Comment by giuseppe** [comment]
I've bumped the cAdvisor commit to 4059641aa7230f5eca4ef1ec6285b2561b97ebfc here: https://github.com/kubernetes/kubernetes/pull/91366

**Pull request closed without merging**